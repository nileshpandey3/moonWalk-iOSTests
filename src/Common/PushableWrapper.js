import React, { Component } from "react";
import { Animated, TouchableOpacity } from "react-native";

export default class extends Component {
  state = {
    pressAnim: new Animated.Value(1),
    appearAnim: new Animated.Value(0)
  };

  componentDidMount() {
    Animated.timing(this.state.appearAnim, {
      toValue: 1,
      duration: 300
    }).start();
  }

  pressAnimation() {
    Animated.timing(this.state.pressAnim, {
      toValue: 0.95,
      duration: 100
    }).start();
  }

  releaseAnimation() {
    Animated.timing(this.state.pressAnim, {
      toValue: 1,
      duration: 100
    }).start();
  }

  render() {
    const { style, children } = this.props;
    const { pressAnim } = this.state;
    return (
      <TouchableOpacity
        onPressIn={() => this.pressAnimation()}
        onPressOut={() => this.releaseAnimation()}
        onPress={() => this.props.onPress()}
        style={style}
      >
        <Animated.View
          style={[
            {
              transform: [{ scale: pressAnim }],
              opacity: this.state.appearAnim
            },
            style
          ]}
        >
          {children}
        </Animated.View>
      </TouchableOpacity>
    );
  }
}
