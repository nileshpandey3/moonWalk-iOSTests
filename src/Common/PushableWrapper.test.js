import React from "react";
import PushableWrapper from "./PushableWrapper";
import renderer from "react-test-renderer";

jest.useFakeTimers();

describe("PushableWrapper", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<PushableWrapper />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
