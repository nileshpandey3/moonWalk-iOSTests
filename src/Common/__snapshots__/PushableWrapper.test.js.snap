// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PushableWrapper renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "opacity": 0,
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
      }
    }
  />
</View>
`;
